generator/src/lib.rs

```diff
@@ -1916,14 +1900,24 @@ pub fn generate_struct(
                 .map(|r| r.to_tokens(field.is_const));
             quote!(#pointer Self)
         } else {
-            field.type_tokens(false)
+            let lifetime = has_lifetimes
+                .contains(&name_to_tokens(&field.basetype))
+                .then(|| quote!(<'a>));
+            let ty = field.type_tokens(false);
+            quote!(#ty #lifetime)
         };
         quote! {pub #param_ident: #param_ty_tokens}
     });
 
-    let debug_tokens = derive_debug(_struct, union_types);
-    let default_tokens = derive_default(_struct);
-    let setter_tokens = derive_setters(_struct, root_structs);
+    let has_lifetime = has_lifetimes.contains(&name);
+    let (generics, marker) = match has_lifetime {
```

Comment:
Bit odd to call it `generics` in one place, `lifetime` in others (there are no generic types involved in this instance too?).